src/pydata_sphinx_theme/toctree.py

```diff
@@ -367,6 +367,11 @@ def add_collapse_checkboxes(soup: BeautifulSoup) -> None:
         if not element.find("ul"):
             continue
 
+        header = soup.new_tag("div", attrs={"class": "toctree-collapsible-header"})
```

Comment:
There are a couple things I don't like with this change.

It creates an inconsistency. TOC entries with children will have the following markup tree:

```
li
  input
  div
    a
    label
  ul
```

Whereas other TOC entries will look like:

```
li
  a
```

There is an additional level of depth in the tree, whereas previously all entries, whether they had children or not, had only one level of tree depth.

I'm worried about two things:

1. Inconsistency (just because inconsistencies like these make me uneasy)
2. Messing up consumers' sites because they customized the theme expecting a direct child relationship between the `li` and the link (or other elements like the chevron)

However, if I don't put the link and the chevron together in a row-flex div, then I don't know how to lay out the link and chevron in the same row without using absolute positioning, which as I mention in another inline comment, creates a hard-to-fix visual aberration.

Reply:
I can see why this is troublesome. I guess you could wrap even the non-expanding headings in a similar div?  As mentioned in another comment I think we shouldn't be *too* fretful about breaking downstream customizations as long as the fix is a straightforward one. Wrapping *all* headings in an extra div seems like it would facilitate those fixes being easier.